ci-operator/step-registry/ipi/conf/vsphere/check/vcm/ipi-conf-vsphere-check-vcm-commands.sh

```diff
@@ -411,31 +393,22 @@ for RESOURCE_POOL in ${RESOURCE_POOLS}; do
   fi
   platformSpec=$(echo "${platformSpec}" | jq -r '.failureDomains += [{"server": "'"${server}"'", "name": "'"${name}"'", "zone": "'"${zone}"'", "region": "'"${region}"'", "server": "'"${server}"'", "topology": {"resourcePool": "'"${resource_pool}"'", "computeCluster": "'"${cluster}"'", "datacenter": "'"${datacenter}"'", "datastore": "'"${datastore}"'", "networks": ["'"${network}"'"]}}]')
 
-  cp /tmp/pool.json "${SHARED_DIR}"/POOL_"${RESOURCE_POOL}".json
-done
-
-log "building vcenters for platform spec"
-
-# append vCenters to platform spec
-for VCENTER in "${!pool_usernames[@]}"; do
-  log "building ${VCENTER} in platform spec"
-  declare -A _datacenters
-  for _poolJSON in "${SHARED_DIR}"/POOL*; do
-    log "processing $_poolJSON";
-    _VCENTER=$(jq -r .spec.server < "${_poolJSON}")
-    _DATACENTER=$(jq -r .spec.topology.datacenter < "${_poolJSON}")
-    if [ "${_VCENTER}" = "${VCENTER}" ]; then
-      _datacenters[${_DATACENTER}]=${_DATACENTER}
-    fi
-  done
-  printf -v joined '"%s",' "${_datacenters[@]}"
-  log "found datacenters ${joined%,}"
+  # Add / Update vCenter list
+  if echo "${platformSpec}" | jq -e --arg VCENTER "$VCENTER" '.vcenters[] | select(.server == $VCENTER) | length > 0' ; then
+    log "Adding additional datacenter to vcenter ${VCENTER}"
+    platformSpec=$(echo "${platformSpec}" | jq -r --arg VCENTER "$VCENTER" --arg DC "$datacenter" '(.vcenters[] | select(.server == $VCENTER)) |= (.datacenters[.datacenters | length] |= . + $DC)')
+  else
+    log "Adding vcenter ${VCENTER} to config"
+    platformSpec=$(echo "${platformSpec}" | jq -r '.vcenters += [{"server": "'"${VCENTER}"'", "user": "'"${pool_usernames[$VCENTER]}"'", "password": "'"${pool_passwords[$VCENTER]}"'", "datacenters": ["'"${datacenter}"'"]}]')
+  fi
 
-  platformSpec=$(echo "${platformSpec}" | jq -r '.vcenters += [{"server": "'"${VCENTER}"'", "user": "'"${pool_usernames[$VCENTER]}"'", "password": "'"${pool_passwords[$VCENTER]}"'", "datacenters": ['"${joined%,}"']}]')
+  cp /tmp/pool.json "${SHARED_DIR}"/POOL_"${RESOURCE_POOL}".json
 done
 
+# For legacy spec, the below will merge the following json to the existing json.
+# TODO: Do we want to just replace / set the vcenter data directly instead?
```

Comment:
>  Do we want to just replace / set the vcenter data directly instead?
yes, that seems reasonable to me.

Reply:
Thanks.  I updated this to just set the platform to the formatted text instead of merging over the version w/ failure domains and vcenters blocks.